### ISSUE TITLE
[Progress bar] My Issue 

### ISSUE THREAD

**Issue by hubble8** [glutanimate/anki-addons-misc#163]
#44  Prerequisite checklist

[ Please enter an [ x ] character to confirm the points below: ]

- [x] In case of a bug: Have you tried restarting Anki?
- [x] Are you running the latest version of the add-on? Have you redownloaded the add-on from AnkiWeb to make sure?
- [x] Did you check the add-on description on AnkiWeb for known issues?
- [x] Did you perform a cursory search through existing issue reports?

## What is the problem/feature you would like to see fixed/implemented?

[ Recently forced to update my anki application to newest version. It would not let me open without it. As result I cannot use progress bar which is my absolute favorite add on. Would love an update. Willing to pay if necessary]

## Version information 2.1.35 

**Anki**

[ Version 2.1.34 ]

**System**

- Operating system: [ macOS ]
- Version: [ Catalina ]

**Comment by imaori** [comment]
Same here.

**Comment by andrest50** [comment]
It seems that the owner of the add-on has not merged pull request glutanimate/anki-addons-misc#152, which should allow you to use the add-on in the newer versions (I'm on 2.1.35, and it works fine). To get the add-on to work, you need to update the reviewer_progress_bar.py file with the updated changes from glutanimate/anki-addons-misc#152. (You can click on the pull request -> click on files changed -> click on the ellipsis and view file in the dropdown to get the entire source code to copy). You'll also need to update the meta.json file so that Anki will stop complaining about the add-on being incompatible with the latest versions. As a user mentioned in the Anki add-on page and below, update the file with this:

{"name": "Progress Bar", "mod": 1504033211, "min_point_version": 0, "max_point_version": -38, "branch_index": 0, "disabled": false, "conflicts": []}

After that, you should be able to run Anki and have the add-on working just fine. The bar is initially tiny (for me at least), so you can always adjust the user preferences within the reviewer_progress_bar.py file to change the styling a bit.

**Comment by kaleo333** [comment]
do you update just the reviewer_progress_bar py file or also the init and meta file? I really have no idea what I'm doing. I have copy and pasted the code from glutanimate/anki-addons-misc#152 but when I load 2.1.35 it said progress bar is not compatible. thank you!

**Comment by andrest50** [comment]
> do you update just the reviewer_progress_bar py file or also the init and meta file? I really have no idea what I'm doing. I have copy and pasted the code from glutanimate/anki-addons-misc#152 but when I load 2.1.35 it said progress bar is not compatible. thank you!

[I updated my previous comment with a better option than what I originally proposed, so, for anyone else with this issue, read that one to fix the add-on].

**Comment by kaleo333** [comment]
> > do you update just the reviewer_progress_bar py file or also the init and meta file? I really have no idea what I'm doing. I have copy and pasted the code from glutanimate/anki-addons-misc#152 but when I load 2.1.35 it said progress bar is not compatible. thank you!
> 
> Speaking from what worked for me, I duplicated the add-on folder for the reviewer progress bar add-on, and then updated the source code for reviewer_progress_bar.py without touching any init or meta file (replaced source code with that of the updated file from PR glutanimate/anki-addons-misc#152). So when I run Anki and go to Tools -> Add-ons, I see both of the add-ons (the original and the updated duplicate). The updated one works fine but the original is still outdated, so I personally just deleted the original. (This is on Windows, though it should work wherever you have your Anki add-ons installed, which is in AppData/Roaming on Windows). Let me know if this works!

I actually changed the meta as someone suggested on the anki add-on page:
{"name": "Progress Bar", "mod": 1504033211, "min_point_version": 0, "max_point_version": -38, "branch_index": 0, "disabled": false, "conflicts": []}

and changed the size settings like you recommended, and it's working perfectly. Doesn't even say it's "disabled" now. thanks!!

**Comment by andrest50** [comment]
Good point, other people should do the same. Great to hear it's working now!

**Comment by kaleo333** [comment]
only problem (that I don't understand) is that if you check for updates for add ons, it changes the meta file back and it goes back to being disabled and you have to swap it out with the edited meta file

**Comment by andrest50** [comment]
> only problem (that I don't understand) is that if you check for updates for add ons, it changes the meta file back and it goes back to being disabled and you have to swap it out with the edited meta file

It seems as though because the add-on is set for versions <= 2.1.26, when you check for updates to add-ons it rewrites the meta.json file to be up to date with what the contents of the file are set to be (by the creator of the add-on when uploading it to the Anki add-ons page). I can't say for certain that it works like this, but either way I see the problem you're having. I just copied the three files from the add-on folder into a new add-on folder and it works fine now. Each folder is an add-on, so you'll end up with two add-ons with the same name of Progress Bar in Anki, but when you check for updates only the original one will become outdated and the new one should be fine. Since you're creating the add-on on your end, it won't update with the meta.json that must be predetermined when checking for updates. (And if you already changed the meta.json again and want to go with this approach to fix the issue, both add-ons will be working initially, so make sure you check for updates and restart Anki so that only the new add-on works with no overlap in the progress bar display.)

**Comment by kaleo333** [comment]
surprisingly simple solution. thank you again. just like you said I had 2 progress bars, checked for updates, restarted, and back to just one. 

I'm also not sure if it's intended functionality from the changes in glutanimate/anki-addons-misc#152, but the progress bar is now always onscreen and never resets when you come back to the main overview page. It also finally only lists your total as your current settings as opposed to every due card and every new card that exists. I prefer the latter change, I just didn't know if I had somehow broken that while updating the addon for 2.1.35

**Comment by andrest50** [comment]
> I'm also not sure if it's intended functionality from the changes in glutanimate/anki-addons-misc#152, but the progress bar is now always onscreen and never resets when you come back to the main overview page. It also finally only lists your total as your current settings as opposed to every due card and every new card that exists. I prefer the latter change, I just didn't know if I had somehow broken that while updating the addon for 2.1.35

I had only just installed the addon prior to updating to 2.1.35 so I'm not so familiar with how it was before, along with all its functionalities, but if it has changed upon updating then it must be as part of the changes. You can always have a look at the user configuration in the source code to see if it helps meet your needs; otherwise, you might have to work with the way it is for now at least. I might have a look more closely at the source code to see if there are some things I can change, but I think what your saying seems to be the way it is designed in the new changes.

**Comment by kaleo333** [comment]
sounds intentional then. I can definitely live with it and some of the changes are even way better. thanks
